File: CoreServices/src/main/java/com/example/entities/Schedule.java
package com.example.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

@Entity
@Table(name = "schedule")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Schedule {
    @Id
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @Column(columnDefinition = "BINARY(16)")
    private UUID id;
    private LocalDate date;
    private LocalTime startTime;
    private LocalTime endTime;
    private boolean available;
    @ManyToOne
    @JoinColumn(name = "doctor_id", nullable = false)
    private Doctor doctor;
    @OneToOne
    @JoinColumn(name = "appointment_id", nullable = true)
    private Appointment appointment;
}

File: GUI/src/app/shared/models/schedule/schedule.ts
import { v4 as uuidv4 } from 'uuid';
import { Doctor } from "../users/doctor/doctor";

export class Schedule {
    public id: string;
    date: string;
    startTime: string;
    endTime: string;
    available: boolean;
    doctorId: string;

    constructor() {
        this.id = uuidv4();
      }
}
